cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/datastore/DatastoreManagerTest.java

```diff
@@ -213,4 +219,73 @@ public void listEmptyDatastore() throws Exception {
         Assert.assertEquals(0, datastores.size());
     }
 
+    @Test
+    public void multithreadedDatastoreCreation() throws Exception {
+        new MultiThreadedTestHelper<Datastore>(25) {
+
+            @Override
+            protected void doAssertions() throws Exception {
+                Datastore ds0 = results.get(0);
+                for (Datastore ds : results) {
+                    Assert.assertSame("The datastore instances should all be the same", ds0, ds);
+                }
+            }
+
+            @Override
+            protected Callable<Datastore> getCallable() {
+                return new
+                        Callable<Datastore>() {
+                            @Override
+                            public Datastore call() throws Exception {
+                                return manager.openDatastore("sameDatastoreName");
+                            }
+                        };
+            }
+        }.run();
+    }
+
+    @Test
+    public void datastoreInstanceNotReusedAfterClose() throws Exception {
+        Datastore ds1 = manager.openDatastore("ds1");
```

Comment:
The `multithreadedDatastoreCreation` test asserts that all the instances are the same, which covers the case of it being reused when it hasn't been closed.